addons/osio-addon/src/main/java/io/fabric8/launcher/osio/client/AuthPublicKeyProvider.java

```diff
@@ -0,0 +1,94 @@
+package io.fabric8.launcher.osio.client;
+
+import java.io.IOException;
+import java.security.interfaces.RSAPublicKey;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.enterprise.context.RequestScoped;
+import javax.inject.Inject;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+import io.fabric8.launcher.base.JsonUtils;
+import io.fabric8.launcher.base.http.HttpClient;
+import io.fabric8.launcher.base.identity.RSAPublicKeyConverter;
+import io.fabric8.launcher.base.identity.TokenIdentity;
+import io.fabric8.launcher.core.spi.Application;
+import io.fabric8.launcher.core.spi.PublicKeyProvider;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+
+import static io.fabric8.kubernetes.client.utils.URLUtils.pathJoin;
+import static io.fabric8.launcher.base.http.HttpClient.getContent;
+import static io.fabric8.launcher.base.http.Requests.securedRequest;
+import static io.fabric8.launcher.osio.OsioConfigs.getAuthUrl;
+import static java.util.Objects.requireNonNull;
+
+@RequestScoped
+@Application(Application.ApplicationType.OSIO)
+public class AuthPublicKeyProvider implements PublicKeyProvider {
```

Comment:
it may be good if we documented public classes

Reply:
What would you like to have mention in the doc what is not quickly understandable from the code itself? Looking for a hint.